src/coordination/coordinator_data.cpp

```diff
@@ -32,25 +33,46 @@ CoordinatorData::CoordinatorData() {
     return *instance;
   };
 
-  replica_succ_cb_ = [find_instance](CoordinatorData *coord_data, std::string_view instance_name) -> void {
+  replica_succ_cb_ = [this, find_instance](CoordinatorData *coord_data, std::string_view instance_name) -> void {
     auto lock = std::lock_guard{coord_data->coord_data_lock_};
     spdlog::trace("Instance {} performing replica successful callback", instance_name);
-    find_instance(coord_data, instance_name).OnSuccessPing();
+    auto &instance = find_instance(coord_data, instance_name);
+
+    if (!instance.IsAlive() || main_uuid_ != instance.GetMainUUID()) [[unlikely]] {
```

Comment:
The instance will for 99.99% of calls be alive, I am confident it is better this way because all other calls will succeed. Expectancy for instance to fail is 1 time in a month. I am okay with removing unlikely, but that was the reasoning.